### DISCUSSION TITLE
Shortcut to render CP or site templates

### DISCUSSION THREAD

**Discussion by sjelfull** [craftcms/cms#8426] [Ideas]
### Description

When plugin devs want to render site templates in a CP request, or plugin templates in a site request, we always have to add boilerplate looking something like this:

```
$view = Craft::$app->getView();
$oldTemplatesPath = $view->getTemplatesPath();

$view->setTemplatesPath(Plugin::getInstance()->getBasePath());
$html = $view->renderTemplate('/bar');

$view->setTemplatesPath($oldTemplatesPath);
```

It would be nice to have some shortcuts like `renderSiteTemplate()` and `renderPluginTemplate()`.

**Comment by khalwat** [comment]
I wrote a method to do exactly that, for exactly the same reason:

https://github.com/nystudio107/craft3-disqus/blob/master/src/services/DisqusService.php#L141

So would support such a thing. :)
